### PR TITLE
feat(pattern/flags): add code and specs docs

### PR DESCRIPTION
**Pull request by tiloyi**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/adeo/design-system--styleguide/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [x] Documentation content changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?

- [ ] Yes
- [ ] No

## Other information

### CODE REVIEW

File: src/pages/Components/Flags/code.md [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+
+## Basic usage
+
+To create a flag you have one main class to apply `mc-flag`.
+
+```html
+<span class="mc-flag">Label</span>
+```
+
+<preview path="src/pages/Components/Flags/previews/flag-default"></preview>
```
[the remark this answers is not in the document]
Comment (caqueste) [reply src/pages/Components/Flags/code.md]: It's OK, well done !

File: src/pages/Components/Flags/code.md [hunk 2]
```diff
@@ -5,37 +5,96 @@ order: 2
...
+$my-map: (
+  'color-theme': (
+    'background': white,
+    'border': blue
+    'text': red
+  )
+);
+```
+
+Now the map of your theme is define you can use the mixin `set--flag-theme`.
```
Comment (marineFabien) [inline src/pages/Components/Flags/code.md]: you want to mean `set-flag-theme` ?

File: src/pages/Components/Flags/code.md [hunk 3]
```diff
@@ -5,37 +5,96 @@ order: 2
...
 ```css
 @import 'settings-tools/_all-settings';
-@import '...';
+@import 'components/_c.flag';
 ```
 
 ## Basic usage
 
-Apply the classes...
+To create a flag you have one main class to apply `mc-flag`, and wrap the text inside the button in a `span.mc-flag__label`. While the text without span will look mostly fine, we use it to opticaly align the characters baseline in the flag.
```
[the remark this answers is not in the document]
Comment (remy115) [reply src/pages/Components/Flags/code.md]: I guess he meant "the box", the component itself when he said "the button" It's not the `button` tag.

File: src/pages/Components/Flags/code.md (hunk 2 above)
Comment (tiloyi) [reply src/pages/Components/Flags/code.md]: It fixed

**Pull request merged**